assets/js/group-details.js

```diff
@@ -0,0 +1,467 @@
+/* global jQuery:false, wpApiSettings:false */
+
+jQuery.ajaxSetup({
+  beforeSend: function(xhr) {
+    xhr.setRequestHeader('X-WP-Nonce', wpApiSettings.nonce);
+  },
+})
+
+jQuery(document).ready(function($) {
+function save_field_api(groupId, post_data){
+  return jQuery.ajax({
+    type:"POST",
+    data:JSON.stringify(post_data),
+    contentType: "application/json; charset=utf-8",
+    dataType: "json",
+    url: wpApiSettings.root + 'dt-hooks/v1/group/'+ groupId,
+  })
+}
+
+function get_group(groupId){
+  return jQuery.ajax({
+    type:"GET",
+    contentType: "application/json; charset=utf-8",
+    dataType: "json",
+    url: wpApiSettings.root + 'dt-hooks/v1/group/'+ groupId
+  })
```

Comment:
I would change this to `return jQuery.ajax({ /* .... */ }).promise()` , just to make it clear that what is returned is a promise, although this should work as well.